lio:
  # Map settings
  map:
    save_map: false
    if_filter: true
    save_map_dir: "map"
    map_name: "map.pcd"
    ds_size: 0.3
    save_interval: 500

  # ROS Topic input
  ros:
    lidar_topic: "/livox/lidar"
    imu_topic: "/livox/imu"

  # Sensor Configuration
  sensor:
    lidar_type: 1
    blind: 0.3
    maxrange: 100.0
    filter_rate: 3.0
    enable_downsample: true
    voxel_fliter_size: 0.5
    gravity_norm: 9.7946
    imu_type: 1
    imu_na: 0.1
    imu_ng: 0.1
    imu_nba: 0.0001
    imu_nbg: 0.0001

  # Extrinsic Parameters
  extrinsic:
    lidar_imu: [-0.011, -0.02329, 0.04412,      # lidar in imu
                1.0, 0.0, 0.0, 
                0.0, 1.0, 0.0, 
                0.0, 0.0, 1.0]
    odom_robo: [0.0, 0, 0.13, 0.0, 0.0, 0.0]


  # Vox hash Map
  hash_map:
    hash_capacity: 100000000
    vox_resolution: 0.5
    insert_resolution: 0.5

  # Occupy Map
  occupy_map:
    occ_resolution: 0.1
    occ_size_x: 60.0
    occ_size_y: 60.0
    occ_size_z: 20.0
    occ_update_range: 20.0
    enable_virtual_wall: true
    slide_update: true

  # Kalman Filter Parameters
  kf:
    kf_type: 0
    kf_max_iterations: 4
    kf_align_gravity: true
    kf_quit_eps: 0.001

  # Submaps
  submap:
    submap_resolution: 0.5
    submap_capacity: 1000000

  # output
  output:
    robot:  false
    plan_env_world: false
    plan_env_body:  false
    ml_map:  false
    planner: false
    map:    true
    dense:  true
    pub_step: 1


  relocation:
    update_map: true
    init_pose: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0] # px, py, pz, roll, pitch, yaw 
    # example: [0.0, 0.0, 1.0, 0.0, 0.0, 90.0]
    # It is recommended to set only the yaw angle.
    # Roll and pitch should be left unset, as they are automatically estimated to maintain level alignment.

  eva:
    timer: true